Sources/RemoteNotification.swift

```diff
@@ -33,28 +33,66 @@ public struct RemoteNotification: CustomStringConvertible, Equatable {
     /// The identifier for the UIUserNotificationCategory associated with the push notification. Derived from userInfo["category"]
     public let categoryIdentifier: String?
     /// Custom fields on the remote notification.
-    public let userInfo: [String : AnyObject]
+    public let userInfo: [String : Any]
     /// The dictionary representation of the remote notification.
-    public let remoteNotificationDictionary: [String : AnyObject]
+    public let remoteNotificationDictionary: [String : Any]
     
-    public init?(remoteNotification: [NSObject : AnyObject]?) {
-        guard let remoteNotification = remoteNotification as? [String : AnyObject] else {
+    // MARK: Equatable
+    
+    public static func ==(lhs: RemoteNotification, rhs: RemoteNotification) -> Bool {
+        if lhs.alert != nil || rhs.alert != nil {
+            guard let lhsAlert = lhs.alert, let rhsAlert = rhs.alert, lhsAlert == rhsAlert else {
+                return false
+            }
+        }
+        
+        if lhs.badge != nil || rhs.badge != nil {
+            guard let lhsBadge = lhs.badge, let rhsBadge = rhs.badge, lhsBadge == rhsBadge else {
+                return false
+            }
+        }
+        
+        if lhs.sound != nil || rhs.sound != nil {
+            guard let lhsSound = lhs.sound, let rhsSound = rhs.sound, lhsSound == rhsSound else {
+                return false
+            }
+        }
+        
+        guard lhs.contentAvailable == rhs.contentAvailable else {
+            return false
+        }
+        
+        if lhs.categoryIdentifier != nil || rhs.categoryIdentifier != nil {
+            guard let lhsCategoryIdentifier = lhs.categoryIdentifier, let rhsCategoryIdentifier = rhs.categoryIdentifier, lhsCategoryIdentifier == rhsCategoryIdentifier else {
+                return false
+            }
+        }
+        
+        guard NSDictionary(dictionary: lhs.userInfo) == NSDictionary(dictionary: rhs.userInfo) else {
+            return false
+        }
+        
+        return true
+    }
+    
+    public init?(remoteNotification: [AnyHashable : Any]?) {
+        guard let remoteNotification = remoteNotification as? [String : Any] else {
             return nil
         }
         
         // Parse the notification payload.
         // See https://developer.apple.com/library/ios/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/Chapters/TheNotificationPayload.html for more information on expected values.
         alert = Alert(remoteNotification: remoteNotification)
         
-        let apnsDictionary = remoteNotification[APSServiceKey] as? [String : AnyObject]
+        let apnsDictionary = remoteNotification[APSServiceKey] as? [String : Any]
         badge = apnsDictionary?[badgeKey] as? Int
```

Comment:
I think these may be safer written as

```
apnsDictionary?[badgeKey] as! Int?
```

Basically that will make it blow up if there's an unexpected type, but also not fail if its nil.  If the cast fails and its not nil, I think we'd want it to blow up.